examples/platform/silabs/BaseApplication.cpp

```diff
@@ -750,15 +790,42 @@ void BaseApplication::ScheduleFactoryReset()
         {
             Provision::Manager::GetInstance().SetProvisionRequired(true);
         }
-        PlatformMgr().HandleServerShuttingDown();
+        PlatformMgr().HandleServerShuttingDown(); // HandleServerShuttingDown calls OnShutdown() which is only implemented for the basic information cluster it seems. And triggers and Event flush, which is not relevant when there are no fabrics left
         ConfigurationMgr().InitiateFactoryReset();
     });
 }
 
+void BaseApplication::DoProvisioningReset()
+{
+    PlatformMgr().ScheduleWork([](intptr_t) {
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
```

Comment:
Can we put this reset code into a method and both DoFactoryReset and this call?

It seems like the same code other than the device reset at the end for the DoFactoryReset